oidc/jwks_test.go

```diff
@@ -138,6 +139,46 @@ func TestMismatchedKeyID(t *testing.T) {
 	testKeyVerify(t, key2, bad, key1, key2)
 }
 
+func TestKeyVerifyContextCanceled(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	payload := []byte("a secret")
+
+	good := newECDSAKey(t)
+	jws, err := jose.ParseSigned(good.sign(t, payload))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	timer := time.NewTimer(3 * time.Second)
```

Comment:
Is there a reason 3 seconds was chosen? Can this block forever instead?

```go
	ch := make(chan struct{})
	s := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		<-ch
	}))
	defer s.Close()
	defer close(ch)
```

Reply:
I generally like to put timeouts so if the test fails, it doesn't somehow end up stuck forever. Happy to make this change though.

Reply:
Yes sorry, and thanks for the change. Tests already have a default timeout and you never now what low powered CI system you're running on. I find it really hard to choose correct numbers here.